### ISSUE TITLE
Switch to JSR-275

### ISSUE THREAD

**Issue by IanMayo** [debrief/limpet#110]
JSR363 is incompatible with GeoTools.

So, we need to switch back to 275, which comes as part of GeoTools.

**Comment by pecko** [comment]
@IanMayo 

Will I create a  PR for the issue63 branch? You can apply it and fix compile errors.

**Comment by IanMayo** [comment]
I've created a new issue (110), where I'll work on the JSR issues.  I'm
going to concentrate on that today & tomorrow.  Yes, you can create a PR
for issue63.

On 4 October 2015 at 15:57, Haris Peco notifications@github.com wrote:

> @IanMayo https://github.com/IanMayo
> 
> Will I create a PR for the issue63 branch? You can apply it and fix
> compile errors.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/debrief/limpet/issues/110#issuecomment-145356079.

## 

Ian Mayo
Deep Blue C Technology Ltd
- New technologies, implemented well

**Comment by pecko** [comment]
The PR: https://github.com/debrief/limpet/pull/111
I haven't applied it, but you can do it and continue to fix this issue.
Look at the TestGeotoolsGeometry.testCreatePoint test. It passes without fixing compile errors, but, of course, we have to fix them. 

And yes, creating this issue is good idea and it is high priority.

**Comment by IanMayo** [comment]
Ok, I've done most of the conversion to JSR 275.  I have just one remaining generics issue - it's related to how we extract the double value of a measurement.

In my branch: **issue110_jar275** there's just this one compile error now - it has beaten me :-(   Well, one error - but it occurs in lots of places...

I'm going to stop for the day now.  I'll restart tomorrow.  

@pecko @dinkoivanov  - if either of you have chance to investigate the compiler error in this branch I'd be really grateful.

I'm back on it tomorrow.

**Comment by pecko** [comment]
I understand. Will check what I can. I probably have to learn measurement framework a little more, but maybe it is some general issue.

**Comment by pecko** [comment]
@IanMayo 

I have fixed all compiler errors. See https://github.com/pecko/limpet/commit/c906833f3fcdb17bbcec662eeaea5c82f2a21d94

Mostly issues is that you used <?> what is "Unknown type". I have replaced it with <Quantity> because these classes are type X<Q extends Quantity>

There are 3 places where I'm not sure. I have added FIXME. I think that they are just the problem with code.
Now, 6 tests fail, but I think that we will solve it in easier way.
